src/cli.rs

```diff
@@ -53,48 +55,85 @@ impl Longest {
             outdated_deps,
             total_deps,
+            theme,
         }
     }
 
     pub fn start(&mut self) -> Result<(), Box<dyn std::error::Error>> {
         enable_raw_mode()?;
-        execute!(self.stdout, Hide)?;
+        queue!(self.stdout, Hide, Clear(ClearType::All))?;
+
+        self.render_header()?;
+        self.render_dependencies(&[])?;
+        self.render_footer_actions()?;
+
+        self.stdout.flush()?;
         Ok(())
     }
 
     pub fn handle_keyboard_event(&mut self) -> Result<Event, Box<dyn std::error::Error>> {
         if let event::Event::Key(key) = event::read()? {
             match (key.code, key.modifiers) {
-                (KeyCode::Up | KeyCode::Left, _) => {
+                (KeyCode::Up | KeyCode::Char('k'), _) => {
+                    let prev_i = self.cursor_location;
                     self.cursor_location = if self.cursor_location == 0 {
                         self.outdated_deps.len() - 1
                     } else {
                         self.cursor_location - 1
                     };
+
+                    self.render_dependencies(&[prev_i, self.cursor_location])?;
                 }
-                (KeyCode::Down | KeyCode::Right, _) => {
+                (KeyCode::Down | KeyCode::Char('j'), _) => {
+                    let prev_i = self.cursor_location;
                     self.cursor_location = (self.cursor_location + 1) % self.outdated_deps.len();
+
+                    self.render_dependencies(&[prev_i, self.cursor_location])?;
+                }
+                (KeyCode::Left | KeyCode::Char('h') | KeyCode::BackTab, _) => {
+                    let prev_i = self.cursor_location;
+
+                    self.cursor_location = self.prev_section_location();
+                    self.render_dependencies(&[prev_i, self.cursor_location])?;
+                }
+                (KeyCode::Right | KeyCode::Char('l') | KeyCode::Tab, _) => {
+                    let prev_i = self.cursor_location;
+
+                    self.cursor_location = self.next_section_location();
+                    self.render_dependencies(&[prev_i, self.cursor_location])?;
                 }
                 (KeyCode::Char(' '), _) => {
                     self.selected[self.cursor_location] = !self.selected[self.cursor_location];
+                    self.render_dependencies(&[self.cursor_location])?;
                 }
                 (KeyCode::Enter, _) => {
                     self.reset_terminal()?;
                     return Ok(Event::UpdateDependencies);
                 }
                 (KeyCode::Char('a'), _) => {
-                    self.selected = vec![true; self.outdated_deps.len()];
+                    if self.selected.iter().all(|s| *s) {
+                        self.selected = vec![false; self.outdated_deps.len()];
+                    } else {
+                        self.selected = vec![true; self.outdated_deps.len()];
+                    }
```

Comment:
This could be simplified to something like:
```suggestion
                    let all_selected = self.selected.iter().all(|s| *s);
                    self.selected = vec![!all_selected; self.outdated_deps.len()];
```